core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -127,16 +130,28 @@ KOKKOS_INLINE_FUNCTION
     std::enable_if_t<std::is_same<T, unsigned long long>::value, T>
         cast_from_half(half_t);
 
-class half_t {
+// use alignas(4) to support 32-bit alignment for reductions
+class alignas(4) half_t {
  public:
   using impl_type = Kokkos::Impl::half_impl_t::type;
 
  private:
   impl_type val;
```

Comment:
I believe you might need an in-class initializer to zero if you default the default constructor.  Not sure if `__half` counts as a builtin-type.  @nliber what do you think?

Reply:
I agree we should add the in-class initializer; `__half` is likely to not be zero initialized.

Reply:
I restored the previous default constructor that uses an initializer list.